### DISCUSSION TITLE
[@types/mithril] Hyperscript interface needs a component override which excludes the State type parameter

### DISCUSSION THREAD

**Discussion by CreaturesInUnitards** [DefinitelyTyped/DefinitelyTyped#68590] [Issues with a @types package]
In certain situations, calling `m(MyClosureComponent, { ... })` will inaccurately show a TS error:

```
Argument of type ClosureComponent<{ your attrs here }> is not assignable to parameter of type string
```

At first, I thought this could be fixed by adding an override `component` block in the Hyperscript interface for Factory/Closure/Class components, which omits the `State` type parameter. Unfortunately this seems to break `attrs` autosuggest for all components

[a comment by an automated account is not shown]

**Comment by dead-claudia** [comment]
What really needs done is the `State` argument needs to be removed entirely from those methods. The value passed to that position should really be `any`.

It's an easy PR to make - just change `types/mithril/index.d.ts`. If anyone wants to take a stab, be my guest. It'll be a bit before I can get to doing it myself, due to IRL concerns, but I can spare some time for review.
